driver-core/src/test/java/com/datastax/driver/core/IndexMetadataTest.java

```diff
@@ -213,15 +217,50 @@ public void should_parse_custom_index_options() {
                 + "USING 'dummy.DummyIndex' WITH OPTIONS = {'foo' : 'bar', 'class_name' : 'dummy.DummyIndex'};", keyspace));
     }
 
-    private ColumnDefinitions.Definition definition(String name, DataType type) {
+    /**
+     * Validates a special case where a 'KEYS' index was created using thrift.  In this particular case the index lacks
+     * index_options, however the index_options value is a 'null' string rather then a null value.
+     *
+     * @test_category metadata
+     * @expected_result Index properly parsed and is present.
+     * @jira_ticket JAVA-834
+     * @since 2.0.11, 2.1.7
+     */
+    @Test(groups = "short")
+    public void should_parse_with_null_string_index_options() {
+        TableMetadata table = getTable("indexing");
+
+        List<ByteBuffer> data = ImmutableList.of(
+                wrap("b@706172656e745f70617468"), // column name 'parent_path'
+                ByteBuffer.allocate(0), // component index (null)
+                wrap("regular"), // kind
+                wrap("cfs_archive_parent_path"), // index name
+                wrap("KEYS"), // index type
+                wrap("org.apache.cassandra.db.marshal.BytesType"), // validator
+                wrap("null") // index options
+        );
+        Row row = ArrayBackedRow.fromData(defs, M3PToken.FACTORY, ProtocolVersion.V3, data);
+        ColumnMetadata column = ColumnMetadata.fromRaw(table, Raw.fromRow(row, VersionNumber.parse("2.1")));
+        IndexMetadata index = column.getIndex();
+
+        assertThat(index).hasName("cfs_archive_parent_path")
+                .isNotKeys() // While the index type is KEYS, since it lacks index_options it does not considered.
```

Comment:
I'm not sure I understand why we loose this information since we have the "index_type" column in the resultset that indicates `KEYS`. Or is it a fake value?
I noticed that for indices created with CQL the index_type column is `COMPOSITES` and index_options is a map containing `{"index_keys":""}` - so maybe we could re-create this map so that this information is not lost?